docs/develop/sdks/nodejs/observability.mdx

```diff
@@ -0,0 +1,138 @@
+
+For Momento specifically, you might desire to capture metrics on the number of requests made, their duration, request or response size, and failure rates. Capture these in the Node.js SDK using a middleware, which intercepts the Momento gRPC calls and responses. Here is an example that uses OpenTelemetry and Prometheus to capture request count faceted by request type:
+
+First, set up metrics in your application:
+```typescript
+const resource = Resource.default().merge(
+    new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: 'Your service name',
+    })
+  );
+
+  // Set up metrics via prometheus
+  const metricsExporter = new PrometheusExporter({});
+
+  const myServiceMeterProvider = new MeterProvider({
+    resource: resource,
+  });
+
+  myServiceMeterProvider.addMetricReader(metricsExporter);
+
+  metrics.setGlobalMeterProvider(myServiceMeterProvider);
+```
+
+Then, create a middleware that captures the metric:
+```typescript
+import {Metadata, StatusObject} from '@grpc/grpc-js';
+import {Middleware, MiddlewareRequestHandler} from '@gomomento/sdk';
+import {metrics} from '@opentelemetry/api';
+import {Counter} from '@opentelemetry/api/build/src/metrics/Metric';
+import {Message} from 'google-protobuf';
+
+class ExampleMetricMiddlewareRequestHandler implements MiddlewareRequestHandler {
+  private requestCounter: Counter;
+  constructor(requestCounter: Counter) {
+    this.requestCounter = requestCounter;
+  }
+  onRequestMetadata(metadata: Metadata): Promise<Metadata> {
+    return Promise.resolve(metadata);
+  }
+
+  onRequestBody(request: Message): Promise<Message> {
+    const requestType = request.constructor.name;
+    this.requestCounter.add(1, {'request.type': requestType});
+    return Promise.resolve(request);
+  }
+
+  onResponseMetadata(metadata: Metadata): Promise<Metadata> {
+    return Promise.resolve(metadata);
+  }
+
+  onResponseBody(response: Message | null): Promise<Message | null> {
+    return Promise.resolve(response);
+  }
+
+  onResponseStatus(status: StatusObject): Promise<StatusObject> {
+    return Promise.resolve(status);
+  }
+}
+
+export class ExampleMetricMiddleware implements Middleware {
+  private readonly requestCounter: Counter;
+  constructor() {
+    const meter = metrics.getMeter('metric-middleware-meter');
+
+    this.requestCounter = meter.createCounter('momento_requests_counter', {
+      description: 'Momento GRPC requests',
+    });
+  }
+
+  onNewRequest(): MiddlewareRequestHandler {
+    return new ExampleMetricMiddlewareRequestHandler(this.requestCounter);
+  }
+}
+```
+
+When you create the Momento CacheClient, add the middleware and the metric will be incremented with each request:
+```typescript
+const momento = new CacheClient({
+    configuration: Configurations.Laptop.v1().withMiddlewares([new ExampleMetricMiddleware()]),
```

Comment:
Does this override the pre-configured middlewares?  (are there any?  I can't remember.)

Reply:
This does override the pre-configured middleware, but there are no defaults. I'll change it to addMiddleware anyway in case we use defaults in the future.